aws-amplify/docs
# Fix additional hydration errors

#### Description of changes:
- After setting up #6237, I ran into more hydration errors related to the gen2 banner when visiting links with hashmarks. This PR updates code in `Layout.tsx` that was using `router.asPath` to use the custom hook that returns `router.asPath` without any hashes so that checks can be made correctly for components like the gen2 banner

Staging site: https://hashmark-hydration-errors.d1ywzrxfkb9wgg.amplifyapp.com/

To test:
1. On your localhost go to http://localhost:3000/javascript/build-a-backend/auth/add-social-provider/#set-up-your-frontend
   - Verify there are no hydration errors

#### Related GitHub issue #, if available:

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?

Which platform(s) are affected by this PR (if applicable)?

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks




      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._
